exporter/signalfxexporter/exporter_test.go

```diff
@@ -1048,3 +1051,296 @@ func TestSignalFxExporterConsumeMetadata(t *testing.T) {
+		config         *Config
+		wantErr        bool
+		wantErrMessage string
+	}{
+		{
+			name: "API CA set",
+			config: &Config{
+				ExporterSettings: config.NewExporterSettings(component.NewID(typeStr)),
+				APIURL:           server.URL,
+				IngestURL:        server.URL,
+				AccessToken:      "random",
+				SyncHostMetadata: true,
+				APITLSSettings: configtls.TLSClientSetting{
+					TLSSetting: configtls.TLSSetting{
+						CAFile: "./testdata/certs/ca.pem",
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "API CA set",
+			config: &Config{
+				ExporterSettings: config.NewExporterSettings(component.NewID(typeStr)),
+				APIURL:           server.URL,
+				IngestURL:        server.URL,
+				AccessToken:      "random",
+				SyncHostMetadata: true,
+			},
+			wantErr:        true,
+			wantErrMessage: "certificate is not trusted",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			observedZapCore, observedLogs := observer.New(zap.DebugLevel)
+			logger := zap.New(observedZapCore)
+			apiTLSCfg, err := tt.config.APITLSSettings.LoadTLSConfig()
+			require.NoError(t, err)
+			serverURL, err := url.Parse(tt.config.APIURL)
+			assert.NoError(t, err)
+			dimClient := dimensions.NewDimensionClient(
+				context.Background(),
+				dimensions.DimensionClientOptions{
+					Token:                 "",
+					APIURL:                serverURL,
+					LogUpdates:            true,
+					Logger:                logger,
+					SendDelay:             1,
+					PropertiesMaxBuffered: 10,
+					MetricsConverter:      *converter,
+					APITLSConfig:          apiTLSCfg,
+				})
+			dimClient.Start()
+
+			se := signalfxExporter{
+				pushMetadata: dimClient.PushMetadata,
+			}
+			sme := signalfMetadataExporter{
+				pushMetadata: se.pushMetadata,
+			}
+
+			err = sme.ConsumeMetadata(metadata)
+			time.Sleep(3 * time.Second)
+			require.NoError(t, err)
+
+			if tt.wantErr {
+				if tt.wantErrMessage != "" {
+					assert.Contains(t, observedLogs.All()[0].Context[0].Interface.(error).Error(), tt.wantErrMessage)
+				}
+			} else {
+				require.Equal(t, 1, observedLogs.Len())
+				require.Nil(t, observedLogs.All()[0].Context[0].Interface)
+			}
+		})
+	}
+}
+
+/*for _, tt := range tests {
+	t.Run(tt.name, func(t *testing.T) {
+		tt.config.APIURL = server.URL
+		sfx, err := newSignalFxExporter(tt.config, logger)
+		assert.NoError(t, err)
+		sfx.pushMetadata(metadata)
+		// We can't control the default send delay of 10 seconds
+		// I am afraid that this is adding significant delay to the test suite
+		// I can make the send delay configurable, or take out this test.
+		time.Sleep(12 * time.Second)
+		if tt.wantErr {
+			if tt.wantErrMessage != "" {
+				assert.Contains(t, observedLogs.All()[0].Context[0].Interface.(error).Error(), tt.wantErrMessage)
+			}
+		} else {
+			require.Equal(t, 0, observedLogs.Len())
+		}
+	})
+}*/
+
+func NewLocalHTTPSTestServer(handler http.Handler) (*httptest.Server, error) {
```

Comment:
Not sure if the API can be actually exposed from `_test` files, but still better to make it not exported.

```suggestion
func newLocalHTTPSTestServer(handler http.Handler) (*httptest.Server, error) {
```

Reply:
thx